collector/hwmon_linux.go

```diff
@@ -0,0 +1,392 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nohwmon
+
+package collector
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	hwMonSubsystem = "hwmon"
+)
+
+type hwMonCollector struct{}
+
+var (
+	invalidMetricChars *regexp.Regexp
+	labelDesc          = []string{"hwmon", "sensor", "label"}
```

Comment:
Please keep in mind that these are package global variables. A more unique name is beneficial.